File: uva00299.py
# bubble sort
# ºâ¥æ´«´X¦¸

def main():
    f = int(input())
    for q in range(f):
        count = 0
        lend = int(input())
        num = list(map(int, input().split()))

        for i in range(lend-1):
            for j in range( lend-i-1):
                if num[j] > num[j+1]:
                    num[j], num[j+1] = num[j+1], num[j]
                    count += 1
        print("Optimal train swapping takes", count, "swaps.")


main()

File: uva00299.cpp
// bubble sort
// ºâ¥æ´«´X¦¸

#include <iostream>
using namespace std;

int main()
{
    

    int f = 0;
    cin >> f;
    for (int ii = 0; ii < f; ii++)
    {
        int arr[50];
        int num = 0;
        int count=0;

        cin >> num;
        for(int i=0;i<num;i++){
            cin>>arr[i];
        }
        for(int i=0;i<num-1;i++){
            for(int j=0;j<num-i-1;j++){
                if(arr[j]>arr[j+1]){
                    swap(arr[j],arr[j+1]);
                    count++;
                }
            }

        }
        cout<<"Optimal train swapping takes "<<count<<" swaps."<<endl;
    }

   
}